src/Analyzers/CSharp/Analyzers/AddAccessibilityModifiers/CSharpAddAccessibilityModifiers.cs

```diff
@@ -33,61 +34,74 @@ public override bool ShouldUpdateAccessibilityModifier(
         if (name.Kind() == SyntaxKind.None)
             return false;
 
+        // User has no preference set.  Do not add or remove any accessibility modifiers.
+        if (option == AccessibilityModifiersRequired.Never)
+            return false;
```

Comment:
caller checks this.  but i found it easier to reason if this code checks all 4 cases itself.